public/components/query_compare/search_result/search_components/search_configs/search_config.tsx

```diff
@@ -53,7 +81,9 @@ export const SearchConfig: FunctionComponent<SearchConfigProps> = ({
       selectIndex: '',
     }));
   };
-
+  
+  const documentIndex = queryNumber == 1? documentsIndexes1: documentsIndexes2
```

Comment:
Idea for refactoring: we could pass down some of these states rather than accessing them through the search relevance context? For example, the `SearchConfig` could just take the parameters `documentIndex` and `pipeline`. This might be a bit more readable and could have performance benefits, since the context is refreshed frequently.

Reply:
documentIndex and pipeline makes sense when there is only one cluster to query from, since it could be two different clusters where the queries are from I have used documentIndex1 and documentIndex2

Reply:
I'm referring to taking in `documentIndex` and `pipeline` as parameters to `SearchConfig` so that `search_configs.tsx` passes them in and we don't need to access the cache. We can retrieve that information (`documentIndexes1` and `documentIndexes2`, etc) elsewhere so we can pass them down directly to the components that need them, eliminating the need to consistently check whether the current component requires query 1 constants or query 2 constants.